packages/server/lib/automation/commands/key_press.ts

```diff
@@ -56,19 +104,32 @@ export const BIDI_VALUE: KeyCodeLookup = {
   'Tab': '\uE004',
 }
 
-export async function bidiKeyPress ({ key }: KeyPressParams, client: Client, context: string, idSuffix?: string): Promise<void> {
+export async function bidiKeyPress ({ key }: KeyPressParams, client: Client, autContext: string, topContext, idSuffix?: string): Promise<void> {
   const value = BIDI_VALUE[key]
 
   if (!value) {
     throw new InvalidKeyError(key)
   }
 
+  const autFrameElement = await client.findElement('css selector', 'iframe.aut-iframe')
+  const activeElement = await client.getActiveElement()
+
+  if (!isEqual(autFrameElement, activeElement)) {
```

Comment:
maybe implied, but wonder if a comment like this might help?
```suggestion
// only send the focus event if the AUT window is not already focused
  if (!isEqual(autFrameElement, activeElement)) {
```

Reply:
I think it's pretty obvious implied here, it's also how the test is written